### PR TITLE
Feature/iteration4

### PR DESCRIPTION
**Pull request by tfettig22**
Completed iteration 4, lock icons can change from unlocked to locked and back on click.  Had to set up another if statement inside our innerHTML for loop so that the lock icons remain in the correct state between clicks of the 'new palette' button.

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /* ********* CSS RESET ********** */
 
-i {
```
Comment (aspitz1) [inline style.css:3]: Can we go ahead and remove these lines (3-5)?

**Review by aspitz1 — APPROVED** [review]
Looks good to me! Thanks Tom.

File: main.js [hunk 2]
```diff
@@ -73,14 +74,45 @@ function displayPalette(event) {
...
       <span>
         <p class="hexcode">${currentPalette.colors[i].hexcode}</p>
-        <i class="lock">🔓</i>
+        <i class="fa-solid fa-lock-open" data-id="${currentPalette.colors[i].id}"></i>
       </span>
     </article>`;
+  } if(currentPalette.colors[i].locked === true) {
+    paletteContainer.innerHTML +=
+    `<article class="color-container" data-id="${currentPalette.colors[i].id}">
+      <div class="color-box color1" style="background:${currentPalette.colors[i].hexcode};"></div>
```
Comment (dstavis) [inline main.js]: color1 class

File: main.js [hunk 3]
```diff
@@ -73,14 +74,45 @@ function displayPalette(event) {
...
     paletteContainer.innerHTML +=
     `<article class="color-container" data-id="${currentPalette.colors[i].id}">
       <div class="color-box color1" style="background:${currentPalette.colors[i].hexcode};"></div>
       <span>
         <p class="hexcode">${currentPalette.colors[i].hexcode}</p>
-        <i class="lock">🔓</i>
+        <i class="fa-solid fa-lock-open" data-id="${currentPalette.colors[i].id}"></i>
       </span>
     </article>`;
+  } if(currentPalette.colors[i].locked === true) {
```
Comment (dstavis) [inline main.js]: could we move this next if down to its own line, instead of having it share a line with the previous if's closing curly brace?

Comment (dstavis) [reply main.js]: actually could this be an else?

File: main.js (hunk 2 above)
Comment (dstavis) [inline main.js]: color1 class

File: main.js [hunk 4]
```diff
@@ -73,14 +74,45 @@ function displayPalette(event) {
...
+      <span>
+        <p class="hexcode">${currentPalette.colors[i].hexcode}</p>
+        <i class="fa-solid fa-lock" data-id="${currentPalette.colors[i].id}"></i>
+      </span>
+    </article>`;
+    }
   }
 }
+
+function lockUnlockColor(event) {
```
Comment (dstavis) [inline main.js:100]: this part took me a while to understand, but it looks good to me

**Comment by aspitz1** [comment]
This is both Tom's and my iterations rolled into one now.

File: main.js [hunk 5]
```diff
@@ -62,20 +62,22 @@ var savePaletteButton = document.querySelector("#save-palette-button");
...
+newPaletteButton.addEventListener("click", handler);
 savePaletteButton.addEventListener("click", savePalette);
 
 var currentPalette = new Palette();
 var savedPalettes = []
 
-function displayPalette(event) {
-  paletteContainer.innerHTML = "";
-
+function handler(event) {
```
Comment (dstavis) [inline main.js:73]: we should rename this

**Comment by aspitz1** [comment]
This PR has been put into iteration 5

**Pull request closed without merging**